pydantic/error_wrappers.py

```diff
@@ -105,13 +105,21 @@ def flatten_errors(
 
 @lru_cache()
 def get_exc_type(cls: Type[Exception]) -> str:
-
-    base_name = 'type_error' if issubclass(cls, TypeError) else 'value_error'
-    if cls in (TypeError, ValueError):
-        # just TypeError or ValueError, no extra code
-        return base_name
-
-    # if it's not a TypeError or ValueError, we just take the lowercase of the exception name
+    type_names = {
+        TypeError: 'type_error',
+        ValueError: 'value_error',
+        AssertionError: 'assertion_error'
+    }
+
+    try:
+        # use bare name
+        return type_names[cls]
+    except KeyError:
+        pass
+
+    # if it's not one of the errors above, we just take the lowercase of the exception name
     # no chaining or snake case logic, use "code" for more complex error types.
-    code = getattr(cls, 'code', None) or cls.__name__.replace('Error', '').lower()
-    return base_name + '.' + code
+    for base_class, base_name in type_names.items():
+        if issubclass(cls, base_class):
```

Comment:
Also even though this class will not be called for an exception except these errors (because validators catch only those), the removal of the else clause causes the function to return nothing if the error is not a subclass. Should I default to `value_error.<code>` as it used to?